### PR TITLE
Add missing timestamps

### PR DESCRIPTION
**Pull request by eanders**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
* Adds `DateCreated` and `DateUpdated` values to new clients added through authoritative data sources in the warehouse
* Migration to find any client who doesn't have a `DateCreated` and set `DateCreated` and optionally `DateUpdated` based on the associated warehouse client record.

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
* Bug fix
* 
## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/controllers/clients_controller.rb [hunk 1]
```diff
@@ -30,6 +30,10 @@ class ClientsController < ApplicationController
 
   def create
     clean_params = client_create_params
+    # Enforce DateCreated and DateUpdated
+    clean_params[:DateCreated] = Time.current
+    clean_params[:DateUpdated] = Time.current
```
Comment (gigxz) [inline app/controllers/clients_controller.rb:35]: These should be getting set automatically from `save` below anyway, right?

File: db/warehouse/migrate/20241112181349_set_missing_timestamps.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class SetMissingTimestamps < ActiveRecord::Migration[7.0]
+  def up
+    # Destination Clients
+    import_batch = []
```
Comment (gigxz) [inline db/warehouse/migrate/20241112181349_set_missing_timestamps.rb]: This should be defined inside the `find_in_batches` loop? With the current setup it looks like clients will be re-imported repeatedly with each batch.

**Review by gigxz — APPROVED** [review]
Logic looks good to me

File: app/controllers/clients_controller.rb (hunk 1 above)
Comment (eanders) [reply app/controllers/clients_controller.rb:35]: That was the root of the issue (they aren't saved automatically on the warehouse side).

File: db/warehouse/migrate/20241112181349_set_missing_timestamps.rb (hunk 2 above)
Comment (eanders) [reply db/warehouse/migrate/20241112181349_set_missing_timestamps.rb]: Good catch, thank you!

**Pull request merged**